opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/instrument.py

```diff
@@ -176,6 +180,47 @@ def record(
             Measurement(amount, self, attributes)
         )
 
+    def time(self):
+        return _Timer(self)
+
+
+class _Timer:
+    def __init__(self, instrument: Histogram):
+        self._instrument = instrument
+        self._attributes: Attributes = {}
+
+    def _new_timer(self):
+        return self.__class__(self._instrument)
+
+    def __enter__(self):
+        # pylint: disable=attribute-defined-outside-init
+        self._start = default_timer()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        duration = max(round((default_timer() - self._start) * 1000), 0)
+        self._instrument.record(duration, self._attributes)
+
+    def set_attribute(self, key: str, value: AttributeValue):
+        self._attributes[key] = value
+
+    def set_attributes(self, attributes: Attributes):
+        if attributes is not None:
+            self._attributes.update(attributes)
```

Comment:
Do we really need these 2 functions? The second one can do the same as the first one and the smallest the API the better.